### DISCUSSION TITLE
Alpine doesn't work for me on IOS (Safari and chrome)

### DISCUSSION THREAD

**Discussion by RoylerMarichal** [alpinejs/alpine#3287] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2805

<div type='discussions-op-text'>

<sup>Originally posted by **cluzstudio** April  4, 2022</sup>
I have a project with the TALL stack without any extra dependencies, it runs perfectly on Android and Windows, on chrome and edge. But in IOS, Safari or chrome alpine doesn't work for me at all... Any suggestions?

**In app layout**
`  <script src="//unpkg.com/alpinejs" defer></script> `
`  @livewireStyles `

 -- Content 

 `@livewireScripts`

**Ej. Menu Alpine**
`<div x-data="{menu:false}">`
...
  ` <button type="button" x-on:click="menu=!menu" >`
...
`  <div x-cloak x-show="menu" >`
...

The website is [https://yavoycuba.com](https://yavoycuba.com)

</div>

**Comment by cbarstow** [comment]
I am experiencing a similar issues here. 
In my case x-on:click or @click events are not working within safari MacOS and iOS, fine in all other browsers across both OS.

**Comment by andreasmatuszewski** [comment]
> I am experiencing a similar issues here. In my case x-on:click or @click events are not working within safari MacOS and iOS, fine in all other browsers across both OS.

Same here. Additionally to that on the newest Chrome Update (on Mac) x:on-click and @click work only sometimes.

**Comment by chrisvidal** [comment]
ah! experiencing same issue with `AlpineJS 3.10.2`
`<script defer src="https://unpkg.com/alpinejs@3.10.2/dist/cdn.min.js"></script>`

**Comment by qichunren** [comment]
Did you fixed the problem?

**Comment by RoylerMarichal** [comment]
> Did you fixed the problem?

No

**Comment by saschalalala** [comment]
@cluzstudio can you create a new example with code that is not working for you? I am working on an alpine/livewire project and could not encounter any such issues. The only things I did encounter:

  - Safari is really weird when it cannot e.g. the google recaptcha api and everything stops working in such a case
  - Safari can filter console.logs, so I first thought I had the same problem as @cbarstow, but when trying alert instead of console.log, everything did work. So I found out about this setting that I have never seen before
![image](https://user-images.githubusercontent.com/3439311/173255636-4a29f1bd-b2a6-4ce9-a34d-95c87c842a33.png)

Safari is Version 15.5 (17613.2.7.1.8), Alpine is 3.10.2

**Comment by chrisvidal** [comment]
didnt fix it, I had to downgrade to AlpineJs 2

**Comment by valeriu-dev** [comment]
@HugoDF @calebporzio

**Comment by saschalalala** [comment]
If anyone of the people with the problem would be able to create a minimal not working example, I would be happy to look into it and help.

I just created a really small index.html that works for me on Safari on both Mac and iOS and I have no idea what's going wrong on your applications:

```html
<html>
<script src="//unpkg.com/alpinejs" defer></script>

<div x-data="{ open: false }">
    <button @click="open = true">Expand</button>

    <span x-show="open">
      Content...
    </span>
</div>

</html>
```

Mentioning for awareness @chrisvidal @andreasmatu @cluzstudio

**Comment by valeriu-dev** [comment]
@saschalalala video demonstration with component that works in Chrome, but not work in Safari

https://user-images.githubusercontent.com/14403446/174564495-9ab7a1a0-609f-4860-801f-4b9eea4b8716.MP4

php packages:
laravel/framework: 9.17.9
livewire/livewire: 2.10.5

js packages:
alpinejs: 3.10.2
laravel-mix: 6.0.49

browsers:
safari: Version 15.5 (17613.2.7.1.8)
chrome: Version 102.0.5005.115 (Official Build) (arm64)

**Comment by valeriu-dev** [comment]
also created an example on codepen: https://codepen.io/vodnicear/pen/ExQzxWK

**Comment by SimoTod** [comment]
> also created an example on codepen: https://codepen.io/vodnicear/pen/ExQzxWK
Your codepen works correctly for me. Safari 16.5
Are you sure you don't have any errors in dev console that would crash Alpine?

**Comment by valeriu-dev** [comment]
@SimoTod yes, console is clean

**Comment by valeriu-dev** [comment]
@SimoTod your test is on iOS or on macOS ?

**Comment by SimoTod** [comment]
MacOS

**Comment by valeriu-dev** [comment]
safari 16.5 ?

<img width="657" alt="image" src="https://user-images.githubusercontent.com/14403446/174574262-fcae16bc-cde7-48ad-891b-483d0a4ee369.png">

**Comment by SimoTod** [comment]
15.5

**Comment by valeriu-dev** [comment]
checked in Safari Technology Preview and also doesn't work
<img width="269" alt="image" src="https://user-images.githubusercontent.com/14403446/174576206-79f525cc-92ce-43ea-88c8-3e218868321f.png">

**Comment by SimoTod** [comment]
https://user-images.githubusercontent.com/8427737/174579706-ed48dcc2-5830-4a25-8017-fbcbcf984a14.mov

**Comment by valeriu-dev** [comment]
solved! :)

problem was in this extensions
<img width="775" alt="image" src="https://user-images.githubusercontent.com/14403446/174581571-f8f48b2d-1269-4777-89e2-10e35fa2c62c.png">

**Comment by mgaerber** [comment]
This issue should be closed already.

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423

**Comment by chatisk** [comment]
I am getting kind of this issue, upon initial reload of the page, I need to do a second click so that the `x-on` emits. after that the x-on event works normally

**Comment by dgknca** [comment]
Alpine v3.13.3 is not working on iOS Safari 13.3 and below too. Am I missing something?

**Reply by ekwoka** [reply]
It is.

What do you find to not be working?

Alpine should have no issues going back to Safari 10 as far as I can tell.

**Reply by dgknca** [reply]
visit this link in iPhone 11 Safari 13 https://alpinejs.dev/component/accordion

<img width="974" alt="Screen Shot 2023-11-27 at 09 17 45" src="https://github.com/alpinejs/alpine/assets/38227845/4436e2c0-aa44-4826-b822-04ca8e8b6db5">

**Reply by ekwoka** [reply]
Unfortunately, it seems Apple doesn't actually want people to support ios 13, as they don't provide any way to actually test things on such an old version of safari. So I don't have any way of validating.

If you can provide the console logs, it might be easier.

I can't identify any features that Alpine uses that would not be supported in that version of safari.

**Reply by dgknca** [reply]
The errors are not helpful. I couldn't find any way to get rid of this, so i need to rewrite my app in jquery.

<img width="350" alt="Screen Shot 2023-11-27 at 10 25 18" src="https://github.com/alpinejs/alpine/assets/38227845/1141a560-cd46-4c1a-bc8f-fb4e98161b6d">

**Reply by ekwoka** [reply]
Why do you need to write it in jquery?

Oh that did help! found the issue!

Optional chaining.

What you can do to solve this is either bundle the package yourself and target something between es2017 and es2019, or instead of the cdn script tag you can do

```js
<script type="module">
  import Alpine from 'https://esm.sh/alpinejs'
  window.Alpine = Alpine
  Alpine.start()
</script>
```

This will automatically target the browser being used, so it will transpile the optional chaining to support the browser.

:+1:

But realistically only 0.4% of users are on ios 13.3 or older, which means they can't really even use most of the internet anyway. Hell, it might be entirely tools for testing sites and not real people.

Apple doesn't support it, so why should you?

**Reply by ekwoka** [reply]
There is a pr #3625 to have the CDN be transpiled back to 2017 for lazy devs, so maybe go voice your support?

**Reply by dgknca** [reply]
> But realistically only 0.4% of users are on ios 13.3 or older, which means they can't really even use most of the internet anyway. Hell, it might be entirely tools for testing sites and not real people.
> 
> Apple doesn't support it, so why should you?

You're absolutely right. If it was my own app I wouldn't care about it, but clients do :)

> Oh that did help! found the issue!
> 
> Optional chaining.

I thought the same thing, but checked [caniuse ](https://caniuse.com/mdn-javascript_operators_conditional). It has a wide support range, I don't think this is related to it.

**Reply by ekwoka** [reply]
Well, it does not support safari 13.3 which is the one you have shown errors from, and the errors point to that issue.

Your link is the wrong feature: https://caniuse.com/mdn-javascript_operators_optional_chaining is the correct one.

>  but clients do 

They need to get smarter about this stuff...they have probably not made a single penny from someone using 4 major version old ios...

But the solution above with esm.sh (or your own bundling) will work wonderfully, as optional chaining is easily transpilable.

**Reply by dgknca** [reply]
Oh sorry, I'm confused. Then your solution will work. Thanks!
